### DISCUSSION TITLE
Ability to purge all feature states for a source

### DISCUSSION THREAD

**Discussion by bothness** [maplibre/maplibre#168] [Cross-project feature proposals]
Currently, there are only two ways I know of to remove the states of all of the features on a source.

1. Keep track of all feature states as they are added (eg. an array of IDs), and remove them individually. This is annoying to code, especially in reactive apps.
2. Remove the source and all its layers, and then re-initialise them. This looks ugly when layers flash off and on, and is also annoying to code in reactive apps.

Life would be much easier is Maplibre simply provided a `map.getSource("id").purgeFeatureStates()` function.

**Comment by gabalis** [comment]
Having a function to target a specific feature in the source would offer a broader usage : one could purge all features, or edit just some of them, or change its coordinates & properties, or remove the feature entirely.  I have no idea if this would actually speedup display as MapLibre might still want to redraw the map from scratch but it would "feel better".  The current way ( getSource, edit one feature out of several thousands, setData ) seems to move huge amounts of data for a single edit. 


`map.getSourceFeatures("featureselectorTBD")`, `map.setSourceFeature()` & `map.removeSourceFeature("featureselectorTBD")` are the first suggestions after 10 seconds of thought. I guess there are better ways.

**Comment by wipfli** [comment]
Thanks for this suggestion. I think it would be best to open an issue in MapLibre GL JS and Native if there is not already one for this issue.
